### PR TITLE
Feat/extcodecopy

### PR DESCRIPTION
**Pull request by Eikix**
- add codecopy
- add slight optimisation on store_padded_segment
- could technically optimize more but it's outside the scope of the PR

## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves: #128 
#464

### CODE REVIEW

File: crates/evm/src/instructions/environmental_information.cairo [hunk 1]
```diff
@@ -191,7 +191,40 @@ impl EnvironmentInformationImpl of EnvironmentInformationTrait {
     /// Copy an account's code to memory
     /// # Specification: https://www.evm.codes/#3c?fork=shanghai
     fn exec_extcodecopy(ref self: Machine) -> Result<(), EVMError> {
-        Result::Ok(())
+        let evm_address = self.stack.pop_eth_address()?;
+        let dest_offset = self.stack.pop_usize()?;
+        let offset = self.stack.pop_usize()?;
+        let size = self.stack.pop_usize()?;
+
+        let maybe_account = AccountTrait::account_at(evm_address)?;
```
Comment (greged93) [inline crates/evm/src/instructions/environmental_information.cairo:199]: should we make an issue for `account_at` to check for balance and bytecode as well?

**Review by greged93 — COMMENTED** [review]
just one comment, lgtm otherwise

File: crates/evm/src/instructions/environmental_information.cairo (hunk 1 above)
Comment (Eikix) [reply crates/evm/src/instructions/environmental_information.cairo:199]: i'm not sure, I don't know if the two aren't equivalent
is it legacy from the EVM? 
I think nonce being 0 should be enough no?

Comment (Eikix) [reply crates/evm/src/instructions/environmental_information.cairo:199]: i don't see a world where an account has nonce == 0 but has a positive balance or a code

Comment (greged93) [reply crates/evm/src/instructions/environmental_information.cairo:199]: I mean the [specs](https://github.com/ethereum/execution-specs/blob/master/src/ethereum/shanghai/fork_types.py#L152) tell us that empty should be nonce 0, no balance and no code. IMO we should follow specs. We will fail on ef-tests anyway if we don't.

Comment (Eikix) [reply crates/evm/src/instructions/environmental_information.cairo:199]: okay let's create an issue

**Review by greged93 — DISMISSED** [review]

**Review by greged93 — APPROVED** [review]
lgtm

**Pull request merged**